src/components/CriteriaProvider.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {
+  createContext,
+  ReactNode,
+  FC,
+  useState,
+  memo,
+  useContext,
+  useCallback
+} from 'react';
+import shortid from 'shortid';
+import { ElmStatement } from 'pathways-model';
+
+interface Criteria {
```

Comment:
I see that there is another `Criteria` interface in `pathways-model.d.ts`... I'm assuming these are different?

Reply:
Yea this is why I opposed the name Criteria originally but nothing better came up. In the sense of this PR all of the "Criteria" is CQL/ELM used for navigating the pathway (transitions and marking nodes as complete). The other "Criteria" which already exists is `pathway.criteria`. This Criteria is for determining the applicability of the pathway